src/nft/components/details/detailsV2/BuyButton.tsx

```diff
@@ -0,0 +1,53 @@
+import { Trans } from '@lingui/macro'
+import { formatNumber } from '@uniswap/conedison/format'
+import { ButtonPrimary } from 'components/Button'
+import Loader from 'components/Icons/LoadingSpinner'
+import { useFetchSingleAsset } from 'nft/hooks/useFetchAssets'
+import { GenieAsset } from 'nft/types'
+import styled from 'styled-components/macro'
+
+import { OfferButton } from './OfferButton'
+import { ButtonStyles } from './shared'
+
+const StyledBuyButton = styled(ButtonPrimary)`
+  display: flex;
+  flex-direction: row;
+  padding: 16px 24px;
+  gap: 8px;
+  line-height: 24px;
+  white-space: nowrap;
+
+  ${ButtonStyles}
+`
+
+const Price = styled.div`
+  color: ${({ theme }) => theme.accentTextLightSecondary};
+`
+
+export const BuyButton = ({ asset, onDataPage }: { asset: GenieAsset; onDataPage?: boolean }) => {
+  const { fetchAndPurchaseSingleAsset, isLoading: isLoadingRoute } = useFetchSingleAsset()
+  const price = asset.sellorders?.[0]?.price.value
+
+  if (!price) {
+    return <OfferButton />
+  }
+
+  return (
+    <>
+      <StyledBuyButton disabled={isLoadingRoute} onClick={() => fetchAndPurchaseSingleAsset(asset)}>
```

Comment:
I am a little confused by this comment it is using a callback - wondering if you could explain more as to what you would like me to do here

Reply:
the onClick function takes a function, which you create new every render. you could use `useCallback` to keep the function referentially equal if you want, but since this is a simple component and unlikely to change, it's a bit of a wash